---
output: github_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# planet :earth_americas:
[![DOI](https://zenodo.org/badge/157781369.svg)](https://zenodo.org/badge/latestdoi/157781369)

`planet` is an R package for inferring ethnicity from placental DNA methylation microarray data [1]. 

## Installation

You can install from this github repo with:

```{r eval = FALSE}
remotes::install_github('wvictor14/planet')
```

*Note: currently, installing with R 3.6.0 results in an warning that can be circumvented with by setting  R_REMOTES_NO_ERRORS_FROM_WARNINGS="true" for the system environment variables #2*

```{r message = FALSE, eval = FALSE}
# Run this if you encounter the above error during install:
withr::with_envvar(
  c(R_REMOTES_NO_ERRORS_FROM_WARNINGS="true"), 
  remotes::install_github('wvictor14/planet')
)
```

## Usage

### Example Data

For demonstration, I use 24 samples from a [placental DNAm dataset from GEO (GSE75196)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE75196) [2], which contains samples 
collected in an Australian population. The DNA methylation data (in betas) can be accessed with `data(pl_betas)` and corresponding sample information from `data(pl_pDat)`. Note that cpgs have been filtered to a random ~10,000 + CpGs used in the models from this package.

```{r, message = F, warning = F}
library(planet) 
library(tidyverse)    

#load example data
data(pl_betas)
data(pl_pDat)

dim(pl_betas)
pl_pDat
```

### Infer ethnicity

**Requirements:**

- CpGs and SNPs that are used in `pl_infer_ethnicity` need to be in your betas `data.frame`. You can ensure that you have all features with the `pl_ethnicity_features` vector:

```{r}
all(pl_ethnicity_features %in% rownames(pl_betas))
```

*Note that if some features are missing, `pl_infer_ethnicity` will throw a warning, but will still work.*

- The betas `data.frame` needs to have *samples in columns* and *CpGs/snps* in rows. The rownames 
mustbe CpG/rs identifiers.
- If you have IDAT files available, then I recommend normalizing your betas `data.frame` using the 
same normalization methods used on the training data:
[**noob**](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3627582/) [3] and 
[**BMIQ**](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3546795/) [4]. To apply these, run
`minfi::preprocessNoob()` on an `rgset` object and then `wateRmelon::BMIQ()`. This has already 
applied to the example data.

```{r}
results <- pl_infer_ethnicity(pl_betas)
print(results, row.names = F)
```

`pl_infer_ethnicity` returns probabilities corresponding to each ethnicity for each sample (e.g 
`Prob_Caucasian`, `Prob_African`, `Prob_Asian`). A final classification is determined in two ways:

1. `Predicted_ethnicity_nothresh` - returns a classification corresponding to the highest 
class-specific probability. 

2. `Predicted_ethnicity` - if the highest class-specific probability is below `0.75`, then the the
sample is assigned an `Amibiguous` label. This threshold can be adjusted with the `threshold` 
argument. Samples with this label might require special attention in downstream analyses.


```{r plot_results, fig.width = 5, fig.height = 2.5, dpi = 300}
results %>%
  ggplot(aes(x = Prob_Caucasian, y = Prob_African, col = Predicted_ethnicity)) +
  geom_point(alpha = 0.7) +
  coord_cartesian(xlim = c(0, 1), ylim = c(0, 1))

results %>%
  ggplot(aes(x = Prob_Caucasian, y = Prob_Asian, col = Predicted_ethnicity)) +
  geom_point(alpha = 0.7) +
  coord_cartesian(xlim = c(0, 1), ylim = c(0, 1))
```

We can't compare this to self-reported ethnicity as it is unavailable. But we know these samples 
were collected in Sydney, Australia, and are therefore likely mostly European with some East Asian 
ancestries. 

```{r}
table(results$Predicted_ethnicity)
```

**A note on adjustment in differential methylation analysis**

Because 'Ambiguous' samples might have different mixtures of ancestries, it might be
inadequate to adjust for them as one group in an analysis of admixed populations (e.g. 50/50 
Asian/African should not be considered the same group as 50/50 Caucasian/African). One solution 
would be to simply remove these samples. Another would be to adjust for the raw probabilities-in 
this case, use only two of the three probabilities, since the third will be redundant 
(probabilities sum to 1). If sample numbers are large enough in each group, stratifying downstream
analyses by ethnicity might also be a valid option.

## References

1. [Yuan V, Price EM, Del Gobbo G, Mostafavi S, Cox B, Binder AM, et al. Accurate ethnicity prediction from placental DNA methylation data. Epigenetics & Chromatin. 2019 Aug 9;12(1):51.](https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-019-0296-3)

2. Yeung KR, Chiu CL, Pidsley R, Makris A, Hennessy A, Lind JM: DNA methylation profiles in preeclampsia and healthy control placentas. Am J Physiol Circ Physiol 2016, 310:H1295–H1303.

3. Triche TJ, Weisenberger DJ, Van Den Berg D, Laird PW, Siegmund KD, Siegmund KD: Low-level processing of Illumina Infinium DNA Methylation BeadArrays. Nucleic Acids Res 2013, 41:e90.

4. Teschendorff AE, Marabita F, Lechner M, Bartlett T, Tegner J, Gomez-Cabrero D, Beck S: A beta-mixture quantile normalization method for correcting probe design bias in Illumina Infinium  450 k DNA methylation data. Bioinformatics 2013, 29:189–96.

