---
output: github_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# planet :earth_americas:
[![DOI](https://zenodo.org/badge/157781369.svg)](https://zenodo.org/badge/latestdoi/157781369)

`planet` is an R package for inferring ethnicity from placental DNA methylation microarray data [1]. 

## Installation

You can install from this github repo with:

```{r message = F, eval = F}
library(devtools)
install_github('wvictor14/planet')
```

## Usage

### Example Data

For demonstration purposes, I downloaded a [placental DNAm dataset from GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE75196) [2], which contains samples 
collected in an Australian population. To save on memory, I only use 8/24 samples, which I have 
saved in this repo as a `minfi::RGChannelSet` object.

```{r, message = F, warning = F}
library(planet)
library(minfi)      # for normalization
library(wateRmelon) # for normalization
library(ggplot2)    

#load example data
data(pl_rgset)
pl_rgset # 8 samples
```

### Preprocessing data

I recommend to normalize your data using the same methods I used to normalize the training 
data. Performance on datasets normalized by other methods has not been evaluated yet. 

If IDATs are supplied, you can apply both `noob`[[3]](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3627582/)
 and `BMIQ`[[4]](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3546795/) normalization. If only
methylated and unmethylated data matrices are available, you can apply just `BMIQ`. 

To apply normalization, run `minfi::preprocessNoob()` and then `wateRmelon::BMIQ()`: 

```{r normalization, message = F, warning = F}
pl_noob <- preprocessNoob(pl_rgset)
pl_bmiq <- BMIQ(pl_noob)
```

Note that `preprocessNoob` will drop SNP probes automatically. Because we need these to infer 
ethnicity, we need to combine the methylation data with the 65 snp probe data (59 SNPs, if using 
EPIC):

```{r combine_betas_snps}
pl_snps <- getSnpBeta(pl_rgset)
pl_dat <- rbind(pl_bmiq, pl_snps)
dim(pl_dat) # 485577     8
```

### Infer ethnicity

The input data needs to contain all 1860 features in the final model. We can check our data for
these features with the `pl_ethnicity_features` vector:

```{r}
all(pl_ethnicity_features %in% rownames(pl_dat))
```

To obtain ethnicity calls, you can supply the full DNA methylation data to `pl_ethnicity_infer()`, 
as long as all  1860 features are present.

```{r}
dim(pl_dat)
results <- pl_infer_ethnicity(pl_dat)
print(results, row.names = F)
```

`pl_infer_ethnicity` returns probabilities corresponding to each ethnicity for each sample (e.g 
`Prob_Caucasian`, `Prob_African`, `Prob_Asian`). A final classification is determined in two ways:

1. `Predicted_ethnicity_nothresh` - returns a classification corresponding to the highest 
class-specific probability. 

2. `Predicted_ethnicity` - if the highest class-specific probability is below `0.75`, then the the
sample is assigned an `Amibiguous` label. This threshold can be adjusted with the `threshold` 
argument. Samples with this label might require special attention in downstream analyses.


```{r plot_results}
qplot(data = results, x = Prob_Caucasian, y = Prob_African, 
     col = Predicted_ethnicity, xlim = c(0,1), ylim = c(0,1))
qplot(data = results, x = Prob_Caucasian, y = Prob_Asian, 
     col = Predicted_ethnicity, xlim = c(0,1), ylim = c(0,1))
```

\*For the entire dataset (not just the subset shown here), 22/24 were predicted Caucasian and 2/24 
Asian.

We can't compare this to self-reported ethnicity as it is unavailable. But we know these samples 
were collected in Sydney, Australia, and are therefore likely mostly European with some East Asian 
ancestries. 

```{r}
table(results$Predicted_ethnicity)
```

### Adjustment in differential methylation analysis

Because 'Ambiguous' samples might have different mixtures of ancestries, it might be
inadequate to adjust for them as one group in an analysis of admixed populations (e.g. 50/50 
Asian/African should not be considered the same group as 50/50 Caucasian/African). One solution 
would be to simply remove these samples. Another would be to adjust for the raw probabilities-in 
this case, use only two of the three probabilities, since the third will be redundant 
(probabilities sum to 1). If sample numbers are large enough in each group, stratifying downstream
analyses by ethnicity might also be a valid option.

## References

[1] Yuan V, Price M, Del Gobbo G, Mostafavi S, Cox B, Binder AM, Michels KB, Marsit C, Robinson W: 
Inferring population structure from placental DNA methylation studies. *In prep.*

[2] Yeung KR, Chiu CL, Pidsley R, Makris A, Hennessy A, Lind JM: DNA methylation profiles in
preeclampsia and healthy control placentas. Am J Physiol Circ Physiol 2016, 310:H1295–H1303.

[3] Triche TJ, Weisenberger DJ, Van Den Berg D, Laird PW, Siegmund KD, Siegmund KD: Low-level
processing of Illumina Infinium DNA Methylation BeadArrays. Nucleic Acids Res 2013, 41:e90.

[4] Teschendorff AE, Marabita F, Lechner M, Bartlett T, Tegner J, Gomez-Cabrero D, Beck S: A
beta-mixture quantile normalization method for correcting probe design bias in Illumina Infinium 
450 k DNA methylation data. Bioinformatics 2013, 29:189–96.

