<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Placenta elastic net DNA methylation ethnicity classifier â€¢ planet</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/journal/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Placenta elastic net DNA methylation ethnicity classifier">
<meta property="og:description" content="This package contains R functions to infer ethnicity/ancestry, gestational age, and cell composition from placental DNA methylation data.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">planet</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/cell_composition.html">Infer Placental Cell Composition</a>
    </li>
    <li>
      <a href="articles/ethnicity.html">Infer Ethnicity</a>
    </li>
    <li>
      <a href="articles/gestational_age.html">Infer Gestational Age</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/wvictor14/planet/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="planet-earth_americas" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#planet-earth_americas" class="anchor"></a>planet :earth_americas:</h1></div>

<p><code>planet</code> is an R package for inferring <strong>ethnicity</strong>, <strong>gestational age</strong>, and <strong>cell composition</strong> from placental DNA methylation data <a href="#references"><span class="math display">1</span><span class="math display">2</span><span class="math display">3</span></a>. See full documentation at <a href="https://wvictor14.github.io/planet/" class="uri">https://wvictor14.github.io/planet/</a></p>
<ul>
<li><p><a href="#installation">Installation</a></p></li>
<li>
<p><a href="#usage">Usage</a></p>
<ul>
<li><a href="#example-data">Example Data</a></li>
<li><a href="#infer-ethnicity">Infer Ethnicity</a></li>
<li><a href="#infer-gestational-age">Infer Gestational Age</a></li>
<li><a href="#infer-cell-composition">Infer Cell Composition</a></li>
</ul>
</li>
<li><p><a href="#references---references-">References</a></p></li>
</ul>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install from this github repo with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'wvictor14/planet'</span><span class="op">)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<div id="example-data" class="section level3">
<h3 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example Data</h3>
<p>For demonstration, I use 24 samples from a placental DNAm dataset from GEO (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE75196">GSE7519</a>) <a href="#references"><span class="math display">4</span></a>, which contains samples collected in an Australian population. The DNA methylation data (in betas) can be accessed with <code><a href="https://rdrr.io/r/utils/data.html">data(pl_betas)</a></code> and corresponding sample information from <code><a href="https://rdrr.io/r/utils/data.html">data(pl_pDat)</a></code>. Note that cpgs have been filtered to a random ~10,000 CpGs, plus the CpGs used in all of the models from this package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://github.com/wvictor14/planet">planet</a></span><span class="op">)</span> 
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="fu">theme_set</span><span class="op">(</span><span class="fu">theme_bw</span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span>

<span class="co">#load example data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">pl_betas</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">pl_pDat</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">pl_betas</span><span class="op">)</span>
<span class="co">#&gt; [1] 13918    24</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pl_pDat</span><span class="op">)</span></code></pre></div>
<div class="kable-table">
<table class="table">
<thead><tr class="header">
<th align="left">sample_id</th>
<th align="left">sex</th>
<th align="left">disease</th>
<th align="right">gestation_wk</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GSM1944936</td>
<td align="left">Male</td>
<td align="left">preeclampsia</td>
<td align="right">36</td>
</tr>
<tr class="even">
<td align="left">GSM1944939</td>
<td align="left">Male</td>
<td align="left">preeclampsia</td>
<td align="right">32</td>
</tr>
<tr class="odd">
<td align="left">GSM1944942</td>
<td align="left">Female</td>
<td align="left">preeclampsia</td>
<td align="right">32</td>
</tr>
<tr class="even">
<td align="left">GSM1944944</td>
<td align="left">Male</td>
<td align="left">preeclampsia</td>
<td align="right">35</td>
</tr>
<tr class="odd">
<td align="left">GSM1944946</td>
<td align="left">Female</td>
<td align="left">preeclampsia</td>
<td align="right">38</td>
</tr>
<tr class="even">
<td align="left">GSM1944948</td>
<td align="left">Female</td>
<td align="left">preeclampsia</td>
<td align="right">36</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="infer-ethnicity" class="section level3">
<h3 class="hasAnchor">
<a href="#infer-ethnicity" class="anchor"></a>Infer Ethnicity</h3>
<p><strong>Requirements:</strong></p>
<ul>
<li>CpGs and SNPs that are used in <code>pl_infer_ethnicity</code> need to be in your betas <code>data.frame</code>. You can ensure that you have all features with the <code>pl_ethnicity_features</code> vector:</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">pl_ethnicity_features</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pl_betas</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p><em>Note that if some features are missing, <code>pl_infer_ethnicity</code> will throw a warning, but will still work.</em></p>
<ul>
<li>The betas <code>data.frame</code> needs to have <em>samples in columns</em> and <em>CpGs/snps</em> in rows. The rownames must be CpG/rs identifiers.</li>
<li>If you have IDAT files available, then I recommend normalizing your betas <code>data.frame</code> using the same normalization methods used on the training data: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3627582/"><strong>noob</strong></a> <a href="#references"><span class="math display">5</span></a> and <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3546795/"><strong>BMIQ</strong></a> <a href="#references"><span class="math display">6</span></a>. To apply these, run <code><a href="https://rdrr.io/pkg/minfi/man/preprocessNoob.html">minfi::preprocessNoob()</a></code> on an <code>rgset</code> object and then <code><a href="https://rdrr.io/pkg/wateRmelon/man/BMIQ.html">wateRmelon::BMIQ()</a></code>. This has already applied to the example data.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pl_infer_ethnicity.html">pl_infer_ethnicity</a></span><span class="op">(</span><span class="va">pl_betas</span><span class="op">)</span>
<span class="co">#&gt; [1] "1860 of 1860 predictors present."</span>

<span class="co"># show last 8 rows</span>
<span class="va">results</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span></code></pre></div>
<div class="kable-table">
<table class="table">
<colgroup>
<col width="9%">
<col width="25%">
<col width="17%">
<col width="11%">
<col width="10%">
<col width="13%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left">Sample_ID</th>
<th align="left">Predicted_ethnicity_nothresh</th>
<th align="left">Predicted_ethnicity</th>
<th align="right">Prob_African</th>
<th align="right">Prob_Asian</th>
<th align="right">Prob_Caucasian</th>
<th align="right">Highest_Prob</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GSM1944959</td>
<td align="left">Asian</td>
<td align="left">Asian</td>
<td align="right">0.0123073</td>
<td align="right">0.9523544</td>
<td align="right">0.0353383</td>
<td align="right">0.9523544</td>
</tr>
<tr class="even">
<td align="left">GSM1944960</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0156961</td>
<td align="right">0.1595213</td>
<td align="right">0.8247827</td>
<td align="right">0.8247827</td>
</tr>
<tr class="odd">
<td align="left">GSM1944961</td>
<td align="left">Asian</td>
<td align="left">Asian</td>
<td align="right">0.0208421</td>
<td align="right">0.8954518</td>
<td align="right">0.0837061</td>
<td align="right">0.8954518</td>
</tr>
<tr class="even">
<td align="left">GSM1944962</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0009276</td>
<td align="right">0.0008801</td>
<td align="right">0.9981923</td>
<td align="right">0.9981923</td>
</tr>
<tr class="odd">
<td align="left">GSM1944963</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0022635</td>
<td align="right">0.0028007</td>
<td align="right">0.9949358</td>
<td align="right">0.9949358</td>
</tr>
<tr class="even">
<td align="left">GSM1944964</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0065973</td>
<td align="right">0.0112013</td>
<td align="right">0.9822014</td>
<td align="right">0.9822014</td>
</tr>
<tr class="odd">
<td align="left">GSM1944965</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0021578</td>
<td align="right">0.0024196</td>
<td align="right">0.9954226</td>
<td align="right">0.9954226</td>
</tr>
<tr class="even">
<td align="left">GSM1944966</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0011397</td>
<td align="right">0.0017651</td>
<td align="right">0.9970952</td>
<td align="right">0.9970952</td>
</tr>
</tbody>
</table>
</div>
<p><code>pl_infer_ethnicity</code> returns probabilities corresponding to each ethnicity for each sample (e.g <code>Prob_Caucasian</code>, <code>Prob_African</code>, <code>Prob_Asian</code>). A final classification is determined in two ways:</p>
<ol>
<li><p><code>Predicted_ethnicity_nothresh</code> - returns a classification corresponding to the highest class-specific probability.</p></li>
<li><p><code>Predicted_ethnicity</code> - if the highest class-specific probability is below <code>0.75</code>, then the the sample is assigned an <code>Amibiguous</code> label. This threshold can be adjusted with the <code>threshold</code> argument. Samples with this label might require special attention in downstream analyses.</p></li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">results</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Prob_Caucasian</span>, y <span class="op">=</span> <span class="va">Prob_African</span>, col <span class="op">=</span> <span class="va">Predicted_ethnicity</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">coord_cartesian</span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_x_continuous</span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_y_continuous</span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'P(Caucasian)'</span>, y <span class="op">=</span> <span class="st">'P(African)'</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-plot_results-1.png" width="100%"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">results</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Prob_Caucasian</span>, y <span class="op">=</span> <span class="va">Prob_Asian</span>, col <span class="op">=</span> <span class="va">Predicted_ethnicity</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">coord_cartesian</span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_x_continuous</span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_y_continuous</span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'P(Caucasian)'</span>, y <span class="op">=</span> <span class="st">'P(Asian)'</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-plot_results-2.png" width="100%"></p>
<p>We canâ€™t compare this to self-reported ethnicity as it is unavailable. But we know these samples were collected in Sydney, Australia, and are therefore likely mostly European with some East Asian ancestries.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">Predicted_ethnicity</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Asian Caucasian </span>
<span class="co">#&gt;         2        22</span></code></pre></div>
<p><strong>A note on adjustment in differential methylation analysis</strong></p>
<p>Because â€˜Ambiguousâ€™ samples might have different mixtures of ancestries, it might be inadequate to adjust for them as one group in an analysis of admixed populations (e.g.Â 50/50 Asian/African should not be considered the same group as 50/50 Caucasian/African). One solution would be to simply remove these samples. Another would be to adjust for the raw probabilities-in this case, use only two of the three probabilities, since the third will be redundant (probabilities sum to 1). If sample numbers are large enough in each group, stratifying downstream analyses by ethnicity might also be a valid option.</p>
</div>
<div id="infer-gestational-age" class="section level3">
<h3 class="hasAnchor">
<a href="#infer-gestational-age" class="anchor"></a>Infer Gestational Age</h3>
<p>There are 3 gestational age clocks for placental DNA methylation data from <a href="#references">Lee Y. et al.Â 2019 <span class="math display">2</span></a>:</p>
<ol>
<li>Robust Placental Clock (RPC)</li>
<li>Control Placental Clock (CPC)</li>
<li>Refined Robust Placental Clock (RRPC)</li>
</ol>
<p>To use each, we can specify the <code>type</code> argument in <code>pl_infer_age</code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We will add this information the sample information data.frame, pl_pDat</span>
<span class="va">pl_pDat</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>ga_RPC <span class="op">=</span> <span class="fu"><a href="reference/pl_infer_age.html">pl_infer_age</a></span><span class="op">(</span><span class="va">pl_betas</span>, type <span class="op">=</span> <span class="st">'RPC'</span><span class="op">)</span>,
         ga_CPC <span class="op">=</span> <span class="fu"><a href="reference/pl_infer_age.html">pl_infer_age</a></span><span class="op">(</span><span class="va">pl_betas</span>, type <span class="op">=</span> <span class="st">'CPC'</span><span class="op">)</span>,
         ga_RRPC <span class="op">=</span> <span class="fu"><a href="reference/pl_infer_age.html">pl_infer_age</a></span><span class="op">(</span><span class="va">pl_betas</span>, type <span class="op">=</span> <span class="st">'RRPC'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  
  <span class="co"># reshape, to plot</span>
  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">contains</span><span class="op">(</span><span class="st">'ga'</span><span class="op">)</span>,
               names_to <span class="op">=</span> <span class="st">'clock_type'</span>,
               names_prefix <span class="op">=</span> <span class="st">'ga_'</span>,
               values_to <span class="op">=</span> <span class="st">'ga'</span><span class="op">)</span> <span class="op">%&gt;%</span>
  
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">gestation_wk</span>, y <span class="op">=</span> <span class="va">ga</span>, col <span class="op">=</span> <span class="va">disease</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, se <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">clock_type</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Reported GA (weeks)'</span>, y <span class="op">=</span> <span class="st">'Inferred GA (weeks)'</span>, col <span class="op">=</span> <span class="st">''</span><span class="op">)</span>
<span class="co">#&gt; [1] "558 of 558 predictors present."</span>
<span class="co">#&gt; [1] "546 of 546 predictors present."</span>
<span class="co">#&gt; [1] "395 of 395 predictors present."</span>
<span class="co">#&gt; `geom_smooth()` using formula 'y ~ x'</span></code></pre></div>
<p><img src="reference/figures/README-pl_infer_age-1.png" width="100%"></p>
<p><em>GA: gestational age</em></p>
</div>
<div id="infer-cell-composition" class="section level3">
<h3 class="hasAnchor">
<a href="#infer-cell-composition" class="anchor"></a>Infer Cell Composition</h3>
<p>To infer cell composition on placental villi DNAm samples, we can use the reference cpgs identified in <span class="math display">3</span>. These are provided in this package as <code>pl_cell_cpgs_third</code> and <code>pl_cell_cpgs_first</code> for third trimester (term) and first trimester samples, respectively.</p>
<p>In this example we are using term villi DNAm data, so we first load the reference cpgs <code>pl_cell_cpgs_third</code>. This is a data frame of 600 cpgs, with mean methylation levels for each cell type.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'pl_cell_cpgs_third'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pl_cell_cpgs_third</span><span class="op">)</span>
<span class="co">#&gt;            Trophoblasts   Stromal  Hofbauer Endothelial      nRBC</span>
<span class="co">#&gt; cg10590657    0.1014098 0.9345796 0.8655285   0.8963641 0.8448382</span>
<span class="co">#&gt; cg14923398    0.1282030 0.8902107 0.9036769   0.9383641 0.9508709</span>
<span class="co">#&gt; cg05348366    0.1305697 0.9519820 0.8803082   0.9065136 0.9278057</span>
<span class="co">#&gt; cg17907628    0.1215249 0.9278777 0.8727841   0.8914412 0.9143601</span>
<span class="co">#&gt; cg26799656    0.1259953 0.9482014 0.8803863   0.8791004 0.9010419</span>
<span class="co">#&gt; cg11862144    0.1561991 0.9430855 0.9114967   0.9341671 0.9647331</span>
<span class="co">#&gt;            Syncytiotrophoblast</span>
<span class="co">#&gt; cg10590657          0.05460441</span>
<span class="co">#&gt; cg14923398          0.05383193</span>
<span class="co">#&gt; cg05348366          0.06546727</span>
<span class="co">#&gt; cg17907628          0.05325227</span>
<span class="co">#&gt; cg26799656          0.06823985</span>
<span class="co">#&gt; cg11862144          0.06044207</span></code></pre></div>
<p>After our reference cpg data is loaded, we can estimate cell composition by applying either the Constrained Projection approach implemented by the R packages <code>minfi</code> or <code>EpiDISH</code>, or a non-constrained approach by <code>EpiDish</code>.</p>
<div id="minfi" class="section level4">
<h4 class="hasAnchor">
<a href="#minfi" class="anchor"></a>Minfi</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hansenlab/minfi">minfi</a></span><span class="op">)</span>

<span class="va">houseman_estimates</span> <span class="op">&lt;-</span> <span class="fu">minfi</span><span class="fu">:::</span><span class="fu">projectCellType</span><span class="op">(</span>
  <span class="va">pl_betas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pl_cell_cpgs_third</span><span class="op">)</span>,<span class="op">]</span>, 
  <span class="va">pl_cell_cpgs_third</span>,
  lessThanOne <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">houseman_estimates</span><span class="op">)</span>
<span class="co">#&gt;            Trophoblasts    Stromal      Hofbauer Endothelial       nRBC</span>
<span class="co">#&gt; GSM1944936    0.1091279 0.04891919  0.000000e+00  0.08983998 0.05294062</span>
<span class="co">#&gt; GSM1944939    0.2299918 0.00000000 -1.806592e-19  0.07888007 0.03374149</span>
<span class="co">#&gt; GSM1944942    0.1934287 0.03483540  0.000000e+00  0.09260353 0.02929310</span>
<span class="co">#&gt; GSM1944944    0.2239896 0.06249135  1.608645e-03  0.11040693 0.04447951</span>
<span class="co">#&gt; GSM1944946    0.1894152 0.07935955  0.000000e+00  0.10587439 0.05407587</span>
<span class="co">#&gt; GSM1944948    0.2045124 0.07657717  0.000000e+00  0.09871149 0.02269798</span>
<span class="co">#&gt;            Syncytiotrophoblast</span>
<span class="co">#&gt; GSM1944936           0.6979477</span>
<span class="co">#&gt; GSM1944939           0.6377822</span>
<span class="co">#&gt; GSM1944942           0.6350506</span>
<span class="co">#&gt; GSM1944944           0.5467642</span>
<span class="co">#&gt; GSM1944946           0.6022329</span>
<span class="co">#&gt; GSM1944948           0.6085825</span></code></pre></div>
</div>
<div id="epidish" class="section level4">
<h4 class="hasAnchor">
<a href="#epidish" class="anchor"></a>EpiDish</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sjczheng/EpiDISH">EpiDISH</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'EpiDISH' was built under R version 4.0.3</span>

<span class="co"># robust partial correlations</span>
<span class="va">epidish_RPC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EpiDISH/man/epidish.html">epidish</a></span><span class="op">(</span>
  beta.m <span class="op">=</span> <span class="va">pl_betas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pl_cell_cpgs_third</span><span class="op">)</span>,<span class="op">]</span>,
  ref.m <span class="op">=</span> <span class="va">pl_cell_cpgs_third</span>,
  method <span class="op">=</span> <span class="st">'RPC'</span><span class="op">)</span>

<span class="co"># CIBERSORT</span>
<span class="va">epidish_CBS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EpiDISH/man/epidish.html">epidish</a></span><span class="op">(</span>
  beta.m <span class="op">=</span> <span class="va">pl_betas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pl_cell_cpgs_third</span><span class="op">)</span>,<span class="op">]</span>,
  ref.m <span class="op">=</span> <span class="va">pl_cell_cpgs_third</span>,
  method <span class="op">=</span> <span class="st">'CBS'</span><span class="op">)</span>

<span class="co"># constrained projection (houseman 2012)</span>
<span class="va">epidish_CP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EpiDISH/man/epidish.html">epidish</a></span><span class="op">(</span>
  beta.m <span class="op">=</span> <span class="va">pl_betas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pl_cell_cpgs_third</span><span class="op">)</span>,<span class="op">]</span>,
  ref.m <span class="op">=</span> <span class="va">pl_cell_cpgs_third</span>,
  method <span class="op">=</span> <span class="st">'CP'</span><span class="op">)</span></code></pre></div>
</div>
<div id="compare" class="section level4">
<h4 class="hasAnchor">
<a href="#compare" class="anchor"></a>Compare</h4>
<p>We can compare the different cell composition estimates.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sjmgarnier/viridis">viridis</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: viridisLite</span>

<span class="fu">bind_rows</span><span class="op">(</span><span class="va">houseman_estimates</span> <span class="op">%&gt;%</span> <span class="va">as.data.frame</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">'CP (Houseman)'</span><span class="op">)</span>,
          <span class="va">epidish_RPC</span><span class="op">$</span><span class="va">estF</span> <span class="op">%&gt;%</span> <span class="va">as.data.frame</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">'RPC'</span><span class="op">)</span>,
          <span class="va">epidish_CBS</span><span class="op">$</span><span class="va">estF</span> <span class="op">%&gt;%</span> <span class="va">as.data.frame</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">'CBS'</span><span class="op">)</span>,
          <span class="va">epidish_CP</span><span class="op">$</span><span class="va">estF</span> <span class="op">%&gt;%</span> <span class="va">as.data.frame</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">'CP (EpiDISH)'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">houseman_estimates</span><span class="op">)</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">algorithm</span>, <span class="va">sample</span><span class="op">)</span>,
               names_to <span class="op">=</span> <span class="st">'component'</span>,
               values_to <span class="op">=</span> <span class="st">'estimate'</span><span class="op">)</span> <span class="op">%&gt;%</span>
  

  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample</span>, y <span class="op">=</span> <span class="va">estimate</span>, fill <span class="op">=</span> <span class="va">component</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">'identity'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">algorithm</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_viridis_d</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_minimal</span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_y_continuous</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.1</span>,<span class="fl">1.1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.5</span>, <span class="fl">1</span><span class="op">)</span>, labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">coord_cartesian</span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">''</span>, fill <span class="op">=</span> <span class="st">''</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-9-1.png" width="100%"></p>
</div>
</div>
</div>
<div id="references-references" class="section level2">
<h2 class="hasAnchor">
<a href="#references-references" class="anchor"></a>References {#references}</h2>
<ol>
<li><p><a href="https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-019-0296-3"><strong>Yuan V</strong>, Price EM, Del Gobbo G, Mostafavi S, Cox B, Binder AM, et al.Â Accurate ethnicity prediction from placental DNA methylation data. Epigenetics &amp; Chromatin. 2019 Aug 9;12(1):51.</a></p></li>
<li><p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6628997/">Lee Y, Choufani S, Weksberg R, Wilson SL, <strong>Yuan V</strong>, et al.Â Placental epigenetic clocks: estimating gestational age using placental DNA methylation levels. Aging (Albany NY). 2019;11(12):4238â€“4253. doi:10.18632/aging.102049</a></p></li>
<li><p><a href="">Yuan V et al.</a></p></li>
<li><p>Yeung KR, Chiu CL, Pidsley R, Makris A, Hennessy A, Lind JM: DNA methylation profiles in preeclampsia and healthy control placentas. Am J Physiol Circ Physiol 2016, 310:H1295â€“H1303.</p></li>
<li><p>Triche TJ, Weisenberger DJ, Van Den Berg D, Laird PW, Siegmund KD, Siegmund KD: Low-level processing of Illumina Infinium DNA Methylation BeadArrays. Nucleic Acids Res 2013, 41:e90.</p></li>
<li><p>Teschendorff AE, Marabita F, Lechner M, Bartlett T, Tegner J, Gomez-Cabrero D, Beck S: A beta-mixture quantile normalization method for correcting probe design bias in Illumina Infinium 450 k DNA methylation data. Bioinformatics 2013, 29:189â€“96.</p></li>
</ol>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="http://github.com/wvictor14/planet/">http://â€‹github.com/â€‹wvictor14/â€‹planet/â€‹</a>
</li>
<li>Report a bug at <br><a href="http://github.com/wvictor14/planet">http://â€‹github.com/â€‹wvictor14/â€‹planet</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Victor Yuan <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://zenodo.org/badge/latestdoi/157781369"><img src="https://zenodo.org/badge/157781369.svg" alt="DOI"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Victor Yuan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
