<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>planet • planet</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="planet">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">planet</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/planet.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/wvictor14/planet/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>planet</h1>
            
      
      <small class="dont-index">Source: <a href="http://github.com/wvictor14/planet/blob/main/vignettes/planet.Rmd" class="external-link"><code>vignettes/planet.Rmd</code></a></small>
      <div class="hidden name"><code>planet.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"planet"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cell-composition">Cell composition<a class="anchor" aria-label="anchor" href="#cell-composition"></a>
</h2>
<p>To infer cell composition on placental villi DNAm samples, we can
need to use placental reference cpgs <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-07186-6" class="external-link">(Yuan
2021)</a>. These are provided in this package as
<code>plCellCpGsThird</code> and <code>plCellCpGsFirst</code> for third
trimester (term) and first trimester samples, respectively.</p>
<p>In this example we are using term villi DNAm data, so we first load
the reference cpgs <code>plCellCpGsThird</code>. This is a data frame of
600 cpgs, with mean methylation levels for each cell type.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cell deconvolution packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hansenlab/minfi" class="external-link">minfi</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sjczheng/EpiDISH" class="external-link">EpiDISH</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># data wrangling and plotting</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://github.com/wvictor14/planet" class="external-link">planet</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load example data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"plBetas"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"plCellCpGsThird"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">plCellCpGsThird</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            Trophoblasts   Stromal  Hofbauer Endothelial      nRBC</span></span>
<span><span class="co">## cg10590657    0.1014098 0.9345796 0.8655285   0.8963641 0.8448382</span></span>
<span><span class="co">## cg14923398    0.1282030 0.8902107 0.9036769   0.9383641 0.9508709</span></span>
<span><span class="co">## cg05348366    0.1305697 0.9519820 0.8803082   0.9065136 0.9278057</span></span>
<span><span class="co">## cg17907628    0.1215249 0.9278777 0.8727841   0.8914412 0.9143601</span></span>
<span><span class="co">## cg26799656    0.1259953 0.9482014 0.8803863   0.8791004 0.9010419</span></span>
<span><span class="co">## cg11862144    0.1561991 0.9430855 0.9114967   0.9341671 0.9647331</span></span>
<span><span class="co">##            Syncytiotrophoblast</span></span>
<span><span class="co">## cg10590657          0.05460441</span></span>
<span><span class="co">## cg14923398          0.05383193</span></span>
<span><span class="co">## cg05348366          0.06546727</span></span>
<span><span class="co">## cg17907628          0.05325227</span></span>
<span><span class="co">## cg26799656          0.06823985</span></span>
<span><span class="co">## cg11862144          0.06044207</span></span></code></pre>
<p>After our reference cpg data is loaded, we can estimate cell
composition by applying either the Constrained Projection approach
implemented by the R packages minfi or EpiDISH, or a non-constrained
approach by EpiDish. I demonstrate how to do both.</p>
<div class="section level4">
<h4 id="minfi">Minfi<a class="anchor" aria-label="anchor" href="#minfi"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">houseman_estimates</span> <span class="op">&lt;-</span> <span class="fu">minfi</span><span class="fu">:::</span><span class="fu">projectCellType</span><span class="op">(</span></span>
<span>  <span class="va">plBetas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">plCellCpGsThird</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>  <span class="va">plCellCpGsThird</span>,</span>
<span>  lessThanOne <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">houseman_estimates</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            Trophoblasts    Stromal     Hofbauer Endothelial       nRBC</span></span>
<span><span class="co">## GSM1944936    0.1091279 0.04891919 0.000000e+00  0.08983998 0.05294062</span></span>
<span><span class="co">## GSM1944939    0.2299918 0.00000000 6.680983e-20  0.07888007 0.03374149</span></span>
<span><span class="co">## GSM1944942    0.1934287 0.03483540 0.000000e+00  0.09260353 0.02929310</span></span>
<span><span class="co">## GSM1944944    0.2239896 0.06249135 1.608645e-03  0.11040693 0.04447951</span></span>
<span><span class="co">## GSM1944946    0.1894152 0.07935955 0.000000e+00  0.10587439 0.05407587</span></span>
<span><span class="co">## GSM1944948    0.2045124 0.07657717 0.000000e+00  0.09871149 0.02269798</span></span>
<span><span class="co">##            Syncytiotrophoblast</span></span>
<span><span class="co">## GSM1944936           0.6979477</span></span>
<span><span class="co">## GSM1944939           0.6377822</span></span>
<span><span class="co">## GSM1944942           0.6350506</span></span>
<span><span class="co">## GSM1944944           0.5467642</span></span>
<span><span class="co">## GSM1944946           0.6022329</span></span>
<span><span class="co">## GSM1944948           0.6085825</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="epidish">EpiDISH<a class="anchor" aria-label="anchor" href="#epidish"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># robust partial correlations</span></span>
<span><span class="va">epidish_RPC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EpiDISH/man/epidish.html" class="external-link">epidish</a></span><span class="op">(</span></span>
<span>  beta.m <span class="op">=</span> <span class="va">plBetas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">plCellCpGsThird</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>  ref.m <span class="op">=</span> <span class="va">plCellCpGsThird</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"RPC"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># CIBERSORT</span></span>
<span><span class="va">epidish_CBS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EpiDISH/man/epidish.html" class="external-link">epidish</a></span><span class="op">(</span></span>
<span>  beta.m <span class="op">=</span> <span class="va">plBetas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">plCellCpGsThird</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>  ref.m <span class="op">=</span> <span class="va">plCellCpGsThird</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"CBS"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1</span></span></code></pre>
<pre><code><span><span class="co">## 2</span></span></code></pre>
<pre><code><span><span class="co">## 3</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># constrained projection (houseman 2012)</span></span>
<span><span class="va">epidish_CP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EpiDISH/man/epidish.html" class="external-link">epidish</a></span><span class="op">(</span></span>
<span>  beta.m <span class="op">=</span> <span class="va">plBetas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">plCellCpGsThird</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>  ref.m <span class="op">=</span> <span class="va">plCellCpGsThird</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"CP"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1</span></span></code></pre>
<pre><code><span><span class="co">## 2</span></span></code></pre>
<pre><code><span><span class="co">## 3</span></span></code></pre>
<pre><code><span><span class="co">## 4</span></span></code></pre>
<pre><code><span><span class="co">## 5</span></span></code></pre>
<pre><code><span><span class="co">## 6</span></span></code></pre>
<pre><code><span><span class="co">## 7</span></span></code></pre>
<pre><code><span><span class="co">## 8</span></span></code></pre>
<pre><code><span><span class="co">## 9</span></span></code></pre>
<pre><code><span><span class="co">## 10</span></span></code></pre>
<pre><code><span><span class="co">## 11</span></span></code></pre>
<pre><code><span><span class="co">## 12</span></span></code></pre>
<pre><code><span><span class="co">## 13</span></span></code></pre>
<pre><code><span><span class="co">## 14</span></span></code></pre>
<pre><code><span><span class="co">## 15</span></span></code></pre>
<pre><code><span><span class="co">## 16</span></span></code></pre>
<pre><code><span><span class="co">## 17</span></span></code></pre>
<pre><code><span><span class="co">## 18</span></span></code></pre>
<pre><code><span><span class="co">## 19</span></span></code></pre>
<pre><code><span><span class="co">## 20</span></span></code></pre>
<pre><code><span><span class="co">## 21</span></span></code></pre>
<pre><code><span><span class="co">## 22</span></span></code></pre>
<pre><code><span><span class="co">## 23</span></span></code></pre>
<pre><code><span><span class="co">## 24</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="compare">Compare<a class="anchor" aria-label="anchor" href="#compare"></a>
</h4>
<p>Below, I demonstrate how we can visually compare the different cell
composition estimates.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"plColors"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># bind estimate data frames and reshape for plotting</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span></span>
<span>  <span class="va">houseman_estimates</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">"CP (Houseman)"</span><span class="op">)</span>,</span>
<span>  <span class="va">epidish_RPC</span><span class="op">$</span><span class="va">estF</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">"RPC"</span><span class="op">)</span>,</span>
<span>  <span class="va">epidish_CBS</span><span class="op">$</span><span class="va">estF</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">"CBS"</span><span class="op">)</span>,</span>
<span>  <span class="va">epidish_CP</span><span class="op">$</span><span class="va">estF</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>algorithm <span class="op">=</span> <span class="st">"CP (EpiDISH)"</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">houseman_estimates</span><span class="op">)</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">algorithm</span>, <span class="va">sample</span><span class="op">)</span>,</span>
<span>    names_to <span class="op">=</span> <span class="st">"component"</span>,</span>
<span>    values_to <span class="op">=</span> <span class="st">"estimate"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  </span>
<span>  <span class="co"># relevel for plot</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>component <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">component</span>,</span>
<span>                            levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>                              <span class="st">"nRBC"</span>, <span class="st">"Endothelial"</span>, <span class="st">"Hofbauer"</span>,</span>
<span>                              <span class="st">"Stromal"</span>, <span class="st">"Trophoblasts"</span>,</span>
<span>                              <span class="st">"Syncytiotrophoblast"</span></span>
<span>                            <span class="op">)</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  </span>
<span>  <span class="co"># plot</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample</span>, y <span class="op">=</span> <span class="va">estimate</span>, fill <span class="op">=</span> <span class="va">component</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">algorithm</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">plColors</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span></span>
<span>    limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.1</span>, <span class="fl">1.1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.5</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>    labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>, fill <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p><img src="planet_files/figure-html/compare_deconvolution-1.png"><!-- --></p>
<p>Some notes:</p>
<ul>
<li>Normalize your data with <code><a href="https://rdrr.io/pkg/minfi/man/preprocessNoob.html" class="external-link">minfi::preprocessNoob</a></code> and
BMIQ</li>
<li>Use all cell CpGs - if some are missing, estimates may vary</li>
<li>If your samples have been processed in a particular manner,
(e.g. sampling from maternal side) expect cell composition to reflect
that</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="gestational-age">Gestational age<a class="anchor" aria-label="anchor" href="#gestational-age"></a>
</h2>
<div class="section level4">
<h4 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h4>
<p>For demonstration, I use 24 samples from a placental DNAm dataset
from GEO, (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE75196" class="external-link">GSE7519</a>),
which contains samples collected in an Australian population. The DNA
methylation data (in betas) can be accessed with
<code>data(plBetas)</code> and corresponding sample information from
<code>data(plPhenoData)</code>. Note that for demonstration purposes,
the cpgs have been filtered to a random ~10,000 CpGs, plus the CpGs used
in all of the functions from this package.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load example data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">plBetas</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">plPhenoData</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">plBetas</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 13918    24</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] 13918    24</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">plPhenoData</span><span class="op">)</span></span></code></pre></div>
<div class="kable-table">
<table class="table">
<colgroup>
<col width="15%">
<col width="9%">
<col width="18%">
<col width="18%">
<col width="12%">
<col width="12%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left">sample_id</th>
<th align="left">sex</th>
<th align="left">disease</th>
<th align="right">gestation_wk</th>
<th align="right">ga_RPC</th>
<th align="right">ga_CPC</th>
<th align="right">ga_RRPC</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GSM1944936</td>
<td align="left">Male</td>
<td align="left">preeclampsia</td>
<td align="right">36</td>
<td align="right">38.46528</td>
<td align="right">38.72867</td>
<td align="right">38.65396</td>
</tr>
<tr class="even">
<td align="left">GSM1944939</td>
<td align="left">Male</td>
<td align="left">preeclampsia</td>
<td align="right">32</td>
<td align="right">33.09680</td>
<td align="right">34.21252</td>
<td align="right">32.62763</td>
</tr>
<tr class="odd">
<td align="left">GSM1944942</td>
<td align="left">Female</td>
<td align="left">preeclampsia</td>
<td align="right">32</td>
<td align="right">34.32520</td>
<td align="right">35.09565</td>
<td align="right">33.32502</td>
</tr>
<tr class="even">
<td align="left">GSM1944944</td>
<td align="left">Male</td>
<td align="left">preeclampsia</td>
<td align="right">35</td>
<td align="right">35.50937</td>
<td align="right">36.71512</td>
<td align="right">35.51295</td>
</tr>
<tr class="odd">
<td align="left">GSM1944946</td>
<td align="left">Female</td>
<td align="left">preeclampsia</td>
<td align="right">38</td>
<td align="right">37.63910</td>
<td align="right">37.57931</td>
<td align="right">36.61721</td>
</tr>
<tr class="even">
<td align="left">GSM1944948</td>
<td align="left">Female</td>
<td align="left">preeclampsia</td>
<td align="right">36</td>
<td align="right">36.77051</td>
<td align="right">38.42632</td>
<td align="right">36.72150</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; # A tibble: 6 x 7</span></span>
<span><span class="co">#&gt;   sample_id  sex   disease   gestation_wk ga_RPC ga_CPC ga_RRPC</span></span>
<span><span class="co">#&gt;   &lt;fct&gt;      &lt;chr&gt; &lt;chr&gt;            &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 GSM1944936 Male  preeclam~           36   38.5   38.7    38.7</span></span>
<span><span class="co">#&gt; 2 GSM1944939 Male  preeclam~           32   33.1   34.2    32.6</span></span>
<span><span class="co">#&gt; 3 GSM1944942 Fema~ preeclam~           32   34.3   35.1    33.3</span></span>
<span><span class="co">#&gt; 4 GSM1944944 Male  preeclam~           35   35.5   36.7    35.5</span></span>
<span><span class="co">#&gt; 5 GSM1944946 Fema~ preeclam~           38   37.6   37.6    36.6</span></span>
<span><span class="co">#&gt; 6 GSM1944948 Fema~ preeclam~           36   36.8   38.4    36.7</span></span></code></pre></div>
<p>There are 3 gestational age clocks for placental DNA methylation data
from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6628997/" class="external-link">(Lee
2019)</a>:</p>
<ol style="list-style-type: decimal">
<li>Robust Placental Clock (RPC)</li>
<li>Control Placental Clock (CPC)</li>
<li>Refined Robust Placental Clock (RRPC)</li>
</ol>
<p>To predict gestational, we load the example data:</p>
<ul>
<li>
<code>plBetas</code> - DNAm data for 24 placental samples</li>
<li>
<code>plPhenoData</code> - Matching sample information</li>
</ul>
</div>
<div class="section level4">
<h4 id="predict-gestational-age">Predict Gestational Age<a class="anchor" aria-label="anchor" href="#predict-gestational-age"></a>
</h4>
<p>To select the type of clock, we can specify the <code>type</code>
argument in <code>predictAge</code>.</p>
<p>We will apply all three clocks on this data, and add the predicted
age to the sample information data.frame, <code>plPhenoData</code>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plPhenoData</span> <span class="op">&lt;-</span> <span class="va">plPhenoData</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    ga_RPC <span class="op">=</span> <span class="fu"><a href="../reference/predictAge.html">predictAge</a></span><span class="op">(</span><span class="va">plBetas</span>, type <span class="op">=</span> <span class="st">"RPC"</span><span class="op">)</span>,</span>
<span>    ga_CPC <span class="op">=</span> <span class="fu"><a href="../reference/predictAge.html">predictAge</a></span><span class="op">(</span><span class="va">plBetas</span>, type <span class="op">=</span> <span class="st">"CPC"</span><span class="op">)</span>,</span>
<span>    ga_RRPC <span class="op">=</span> <span class="fu"><a href="../reference/predictAge.html">predictAge</a></span><span class="op">(</span><span class="va">plBetas</span>, type <span class="op">=</span> <span class="st">"RRPC"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 558 of 558 predictors present.</span></span></code></pre>
<pre><code><span><span class="co">## 546 of 546 predictors present.</span></span></code></pre>
<pre><code><span><span class="co">## 395 of 395 predictors present.</span></span></code></pre>
<p>Note that the number of predictors (CpGs) that were used in our data
are printed. It’s important to take note if a significant number of
predictive CpGs are missing in your data, as this can affect the
predicted gestational age accuracy.</p>
<p>Next, I plot the difference between predicted and reported
gestational age, for each of the 3 gestational age predictors.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plPhenoData</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  </span>
<span>  <span class="co"># reshape, to plot</span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">"ga"</span><span class="op">)</span>,</span>
<span>    names_to <span class="op">=</span> <span class="st">"clock_type"</span>,</span>
<span>    names_prefix <span class="op">=</span> <span class="st">"ga_"</span>,</span>
<span>    values_to <span class="op">=</span> <span class="st">"ga"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  </span>
<span>  <span class="co"># plot code</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">gestation_wk</span>, y <span class="op">=</span> <span class="va">ga</span>, col <span class="op">=</span> <span class="va">disease</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html" class="external-link">geom_smooth</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"lm"</span>, se <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">clock_type</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Reported GA (weeks)"</span>, y <span class="op">=</span> <span class="st">"Inferred GA (weeks)"</span>, col <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `geom_smooth()` using formula = 'y ~ x'</span></span></code></pre>
<p><img src="planet_files/figure-html/view_ga-1.png"><!-- --></p>
</div>
</div>
<div class="section level2">
<h2 id="ethnicity">Ethnicity<a class="anchor" aria-label="anchor" href="#ethnicity"></a>
</h2>
<p>Before predicting ethnicity You can ensure that you have all features
using the <code>ethnicityCpGs</code> vector:</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ethnicityCpGs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">ethnicityCpGs</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">plBetas</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictEthnicity.html">predictEthnicity</a></span><span class="op">(</span><span class="va">plBetas</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1860 of 1860 predictors present.</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<div class="kable-table">
<table class="table">
<colgroup>
<col width="9%">
<col width="25%">
<col width="17%">
<col width="11%">
<col width="9%">
<col width="13%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left">Sample_ID</th>
<th align="left">Predicted_ethnicity_nothresh</th>
<th align="left">Predicted_ethnicity</th>
<th align="right">Prob_African</th>
<th align="right">Prob_Asian</th>
<th align="right">Prob_Caucasian</th>
<th align="right">Highest_Prob</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GSM1944959</td>
<td align="left">Asian</td>
<td align="left">Asian</td>
<td align="right">0.0123073</td>
<td align="right">0.9523544</td>
<td align="right">0.0353383</td>
<td align="right">0.9523544</td>
</tr>
<tr class="even">
<td align="left">GSM1944960</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0156961</td>
<td align="right">0.1595213</td>
<td align="right">0.8247827</td>
<td align="right">0.8247827</td>
</tr>
<tr class="odd">
<td align="left">GSM1944961</td>
<td align="left">Asian</td>
<td align="left">Asian</td>
<td align="right">0.0208421</td>
<td align="right">0.8954518</td>
<td align="right">0.0837061</td>
<td align="right">0.8954518</td>
</tr>
<tr class="even">
<td align="left">GSM1944962</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0009276</td>
<td align="right">0.0008801</td>
<td align="right">0.9981923</td>
<td align="right">0.9981923</td>
</tr>
<tr class="odd">
<td align="left">GSM1944963</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0022635</td>
<td align="right">0.0028007</td>
<td align="right">0.9949358</td>
<td align="right">0.9949358</td>
</tr>
<tr class="even">
<td align="left">GSM1944964</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0065973</td>
<td align="right">0.0112013</td>
<td align="right">0.9822014</td>
<td align="right">0.9822014</td>
</tr>
<tr class="odd">
<td align="left">GSM1944965</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0021578</td>
<td align="right">0.0024196</td>
<td align="right">0.9954226</td>
<td align="right">0.9954226</td>
</tr>
<tr class="even">
<td align="left">GSM1944966</td>
<td align="left">Caucasian</td>
<td align="left">Caucasian</td>
<td align="right">0.0011397</td>
<td align="right">0.0017651</td>
<td align="right">0.9970952</td>
<td align="right">0.9970952</td>
</tr>
</tbody>
</table>
</div>
<p><code>predictEthnicity</code> returns probabilities corresponding to
each ethnicity for each sample (e.g <code>Prob_Caucasian</code>,
<code>Prob_African</code>, <code>Prob_Asian</code>). This applies a
glmnet model described in <a href="https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-019-0296-3" class="external-link">(Yuan
2019)</a>. A final classification is determined in two ways:</p>
<ol style="list-style-type: decimal">
<li><p><code>Predicted_ethnicity_nothresh</code> - returns a
classification corresponding to the highest class-specific
probability.</p></li>
<li><p><code>Predicted_ethnicity</code> - if the highest class-specific
probability is below <code>0.75</code>, then the the sample is assigned
an <code>Amibiguous</code> label. This threshold can be adjusted with
the <code>threshold</code> argument. Samples with this label might
require special attention in downstream analyses.</p></li>
</ol>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">Prob_Caucasian</span>, y <span class="op">=</span> <span class="va">Prob_African</span>,</span>
<span>    col <span class="op">=</span> <span class="va">Predicted_ethnicity</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"P(Caucasian)"</span>, y <span class="op">=</span> <span class="st">"P(African)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="planet_files/figure-html/unnamed-chunk-2-1.png"><!-- --></p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">Prob_Caucasian</span>, y <span class="op">=</span> <span class="va">Prob_Asian</span>,</span>
<span>    col <span class="op">=</span> <span class="va">Predicted_ethnicity</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/percent_format.html" class="external-link">percent</a></span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"P(Caucasian)"</span>, y <span class="op">=</span> <span class="st">"P(Asian)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="planet_files/figure-html/unnamed-chunk-2-2.png"><!-- --></p>
<p>We can’t compare this to self-reported ethnicity as it is
unavailable. But we know these samples were collected in Sydney,
Australia, and are therefore likely mostly European with some East Asian
participants.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">Predicted_ethnicity</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##     Asian Caucasian </span></span>
<span><span class="co">##         2        22</span></span></code></pre>
<p><strong>A note on adjustment in differential methylation
analysis</strong></p>
<p>Because ‘Ambiguous’ samples might have different mixtures of
ancestries, it might be inadequate to adjust for them as one group in an
analysis of admixed populations (e.g. 50/50 Asian/African should not be
considered the same group as 50/50 Caucasian/African). One solution
would be to simply remove these samples. Another would be to adjust for
the raw probabilities-in this case, use only two of the three
probabilities, since the third will be redundant (probabilities sum to
1). If sample numbers are large enough in each group, stratifying
downstream analyses by ethnicity might also be a valid option.</p>
</div>
<div class="section level2">
<h2 id="early-onset-preeclampsia-eope">Early-onset preeclampsia (EOPE)<a class="anchor" aria-label="anchor" href="#early-onset-preeclampsia-eope"></a>
</h2>
<p>To calculate the probability of EOPE of placental DNAm chorionic
villi samples, we rely on 45 predictive CpGs.</p>
<p>In this example, we load the validation data used in <a href="https://www.medrxiv.org/content/10.1101/2023.05.17.23290125v1" class="external-link">Fernández-Boyano
2023</a>] and estimate the EOPE probability of the samples. Note that
the function must be run on a matrix with the full set of CpG probes in
either the 450K or 850K arrays - the reason for this is that all 45
predictive CpGs must be present for prediction to be completed.</p>
<p>It is recommended that data is normalized using BMIQ prior to
prediction.</p>
<p>Note that samples must be rows and CpGs must be columns. The default
threshold for classification used to assign labels is 55%; if the users
wishes to use other threshold, different labels can be assigned based on
the output probabilities.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationHub</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocFileCache</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: dbplyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dbplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     ident, sql</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'AnnotationHub'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     cache</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="st">"eoPredData"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ExperimentHub with 3 records</span></span>
<span><span class="co">## # snapshotDate(): 2024-10-24</span></span>
<span><span class="co">## # $dataprovider: University of British Columbia</span></span>
<span><span class="co">## # $species: Homo sapiens</span></span>
<span><span class="co">## # $rdataclass: mixo_splsda, matrix, data.frame</span></span>
<span><span class="co">## # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">## #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">## # retrieve records with, e.g., 'object[["EH8090"]]' </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##            title      </span></span>
<span><span class="co">##   EH8090 | eoPredModel</span></span>
<span><span class="co">##   EH8403 | valBMIQ    </span></span>
<span><span class="co">##   EH8404 | valMeta</span></span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download BMIQ normalized 450k data</span></span>
<span><span class="va">x_test</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">'EH8403'</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## eoPredData not installed.</span></span>
<span><span class="co">##   Full functionality, documentation, and loading of data might not be possible without installing</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">preds</span> <span class="op">&lt;-</span> <span class="va">x_test</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="../reference/predictPreeclampsia.html">predictPreeclampsia</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## eoPredData not installed.</span></span>
<span><span class="co">##   Full functionality, documentation, and loading of data might not be possible without installing</span></span>
<span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## 45 of 45 predictive CpGs present.</span></span></code></pre>
<pre><code><span><span class="co">## BMIQ normalization is recommended for best results. If choosing other method, it is recommended to compare results to predictions on BMIQ normalized data.</span></span></code></pre>
<p>Inspect the results:</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span></span></code></pre></div>
<div class="kable-table">
<table class="table">
<thead><tr class="header">
<th align="left">Sample_ID</th>
<th align="right">EOPE</th>
<th align="right">Non-PE Preterm</th>
<th align="left">PE_Status</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GSM2589533</td>
<td align="right">0.6699478</td>
<td align="right">0.3300522</td>
<td align="left">EOPE</td>
</tr>
<tr class="even">
<td align="left">GSM2589535</td>
<td align="right">0.7680531</td>
<td align="right">0.2319469</td>
<td align="left">EOPE</td>
</tr>
<tr class="odd">
<td align="left">GSM2589536</td>
<td align="right">0.8068395</td>
<td align="right">0.1931605</td>
<td align="left">EOPE</td>
</tr>
<tr class="even">
<td align="left">GSM2589538</td>
<td align="right">0.7843615</td>
<td align="right">0.2156385</td>
<td align="left">EOPE</td>
</tr>
<tr class="odd">
<td align="left">GSM2589540</td>
<td align="right">0.3858278</td>
<td align="right">0.6141722</td>
<td align="left">Normotensive</td>
</tr>
<tr class="even">
<td align="left">GSM2589541</td>
<td align="right">0.6488529</td>
<td align="right">0.3511471</td>
<td align="left">EOPE</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># join with metadata</span></span>
<span><span class="va">valMeta</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">'EH8404'</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## eoPredData not installed.</span></span>
<span><span class="co">##   Full functionality, documentation, and loading of data might not be possible without installing</span></span></code></pre>
<pre><code><span><span class="co">## downloading 1 resources</span></span></code></pre>
<pre><code><span><span class="co">## retrieving 1 resource</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">valMeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">valMeta</span>, <span class="va">preds</span>, by<span class="op">=</span><span class="st">"Sample_ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize results</span></span>
<span><span class="va">plot_predictions</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">df</span>, <span class="va">predicted_class_column</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Class</span>, y <span class="op">=</span> <span class="va">EOPE</span><span class="op">)</span>,</span>
<span>      width <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>      alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>      color <span class="op">=</span> <span class="st">"darkgrey"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Class</span>, y <span class="op">=</span> <span class="va">EOPE</span>, color <span class="op">=</span> <span class="op">{</span><span class="op">{</span><span class="va">predicted_class_column</span><span class="op">}</span><span class="op">}</span><span class="op">)</span>, </span>
<span>      alpha <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>      position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Class Probability of EOPE"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"True Class"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span></span>
<span>      name <span class="op">=</span> <span class="st">"Predicted Class"</span>, </span>
<span>      values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#E69F00"</span>, <span class="st">"skyblue"</span>, <span class="st">"#999999"</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">valMeta</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plot_predictions</span><span class="op">(</span><span class="va">PE_Status</span><span class="op">)</span></span></code></pre></div>
<p><img src="planet_files/figure-html/unnamed-chunk-5-1.png"><!-- --></p>
<p>if user wishes to use different threshold from 55% probability, as an
example 70%</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">valMeta</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>  Pred_Class <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span></span>
<span>    <span class="va">EOPE</span> <span class="op">&gt;</span> <span class="fl">0.7</span> <span class="op">~</span> <span class="st">"EOPE"</span>,</span>
<span>    <span class="va">`Non-PE Preterm`</span> <span class="op">&gt;</span> <span class="fl">0.7</span> <span class="op">~</span> <span class="st">"Non-PE Preterm"</span>,</span>
<span>    .default <span class="op">=</span> <span class="st">'low-confidence'</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">plot_predictions</span><span class="op">(</span><span class="va">Pred_Class</span><span class="op">)</span></span></code></pre></div>
<p><img src="planet_files/figure-html/unnamed-chunk-6-1.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-07186-6" class="external-link">Yuan
V, Hui D, Yin Y, Peñaherrera MS, Beristain AG, Robinson WP.
Cell-specific characterization of the placental methylome. BMC Genomics.
2021 Jan 6;22(1):6.</a></p>
<p><a href="https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-019-0296-3" class="external-link">Yuan
V, Price EM, Del Gobbo G, Mostafavi S, Cox B, Binder AM, et al. Accurate
ethnicity prediction from placental DNA methylation data. Epigenetics
&amp; Chromatin. 2019 Aug 9;12(1):51.</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6628997/" class="external-link">Lee
Y, Choufani S, Weksberg R, Wilson SL, Yuan V, Burt A, et al. Placental
epigenetic clocks: estimating gestational age using placental DNA
methylation levels. Aging (Albany NY). 2019 Jun
24;11(12):4238–53.</a></p>
<p><a href="https://www.medrxiv.org/content/10.1101/2023.05.17.23290125v1" class="external-link"><strong>Fernández-Boyano
I</strong>, A.M. Inkster, <strong>V. Yuan</strong>, W.P. Robinson
medRxiv 2023 May</a></p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ExperimentHub_2.14.0        AnnotationHub_3.14.0       </span></span>
<span><span class="co">##  [3] BiocFileCache_2.14.0        dbplyr_2.5.0               </span></span>
<span><span class="co">##  [5] planet_1.15.4               tidyr_1.3.1                </span></span>
<span><span class="co">##  [7] ggplot2_3.5.1               dplyr_1.1.4                </span></span>
<span><span class="co">##  [9] EpiDISH_2.22.0              minfi_1.52.1               </span></span>
<span><span class="co">## [11] bumphunter_1.48.0           locfit_1.5-9.10            </span></span>
<span><span class="co">## [13] iterators_1.0.14            foreach_1.5.2              </span></span>
<span><span class="co">## [15] Biostrings_2.74.1           XVector_0.46.0             </span></span>
<span><span class="co">## [17] SummarizedExperiment_1.36.0 Biobase_2.66.0             </span></span>
<span><span class="co">## [19] MatrixGenerics_1.18.1       matrixStats_1.5.0          </span></span>
<span><span class="co">## [21] GenomicRanges_1.58.0        GenomeInfoDb_1.42.3        </span></span>
<span><span class="co">## [23] IRanges_2.40.1              S4Vectors_0.44.0           </span></span>
<span><span class="co">## [25] BiocGenerics_0.52.0        </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.4.2             BiocIO_1.16.0            </span></span>
<span><span class="co">##   [3] bitops_1.0-9              filelock_1.0.3           </span></span>
<span><span class="co">##   [5] tibble_3.2.1              preprocessCore_1.68.0    </span></span>
<span><span class="co">##   [7] XML_3.99-0.18             lifecycle_1.0.4          </span></span>
<span><span class="co">##   [9] lattice_0.22-6            MASS_7.3-61              </span></span>
<span><span class="co">##  [11] base64_2.0.2              scrime_1.3.5             </span></span>
<span><span class="co">##  [13] magrittr_2.0.3            limma_3.62.2             </span></span>
<span><span class="co">##  [15] sass_0.4.9                rmarkdown_2.29           </span></span>
<span><span class="co">##  [17] jquerylib_0.1.4           yaml_2.3.10              </span></span>
<span><span class="co">##  [19] doRNG_1.8.6.1             askpass_1.2.1            </span></span>
<span><span class="co">##  [21] DBI_1.2.3                 RColorBrewer_1.1-3       </span></span>
<span><span class="co">##  [23] abind_1.4-8               zlibbioc_1.52.0          </span></span>
<span><span class="co">##  [25] quadprog_1.5-8            purrr_1.0.2              </span></span>
<span><span class="co">##  [27] RCurl_1.98-1.16           rappdirs_0.3.3           </span></span>
<span><span class="co">##  [29] GenomeInfoDbData_1.2.13   ggrepel_0.9.6            </span></span>
<span><span class="co">##  [31] rentrez_1.2.3             genefilter_1.88.0        </span></span>
<span><span class="co">##  [33] ellipse_0.5.0             RSpectra_0.16-2          </span></span>
<span><span class="co">##  [35] annotate_1.84.0           DelayedMatrixStats_1.28.1</span></span>
<span><span class="co">##  [37] codetools_0.2-20          DelayedArray_0.32.0      </span></span>
<span><span class="co">##  [39] xml2_1.3.6                tidyselect_1.2.1         </span></span>
<span><span class="co">##  [41] locfdr_1.1-8              UCSC.utils_1.2.0         </span></span>
<span><span class="co">##  [43] farver_2.1.2              beanplot_1.3.1           </span></span>
<span><span class="co">##  [45] illuminaio_0.48.0         GenomicAlignments_1.42.0 </span></span>
<span><span class="co">##  [47] jsonlite_1.8.9            multtest_2.62.0          </span></span>
<span><span class="co">##  [49] e1071_1.7-16              survival_3.7-0           </span></span>
<span><span class="co">##  [51] tools_4.4.2               Rcpp_1.0.14              </span></span>
<span><span class="co">##  [53] glue_1.8.0                rARPACK_0.11-0           </span></span>
<span><span class="co">##  [55] gridExtra_2.3             SparseArray_1.6.1        </span></span>
<span><span class="co">##  [57] xfun_0.50                 mixOmics_6.30.0          </span></span>
<span><span class="co">##  [59] mgcv_1.9-1                HDF5Array_1.34.0         </span></span>
<span><span class="co">##  [61] withr_3.0.2               BiocManager_1.30.25      </span></span>
<span><span class="co">##  [63] fastmap_1.2.0             rhdf5filters_1.18.0      </span></span>
<span><span class="co">##  [65] openssl_2.3.1             digest_0.6.37            </span></span>
<span><span class="co">##  [67] R6_2.5.1                  mime_0.12                </span></span>
<span><span class="co">##  [69] colorspace_2.1-1          RSQLite_2.3.9            </span></span>
<span><span class="co">##  [71] generics_0.1.3            data.table_1.16.4        </span></span>
<span><span class="co">##  [73] corpcor_1.6.10            rtracklayer_1.66.0       </span></span>
<span><span class="co">##  [75] class_7.3-22              httr_1.4.7               </span></span>
<span><span class="co">##  [77] htmlwidgets_1.6.4         S4Arrays_1.6.0           </span></span>
<span><span class="co">##  [79] pkgconfig_2.0.3           gtable_0.3.6             </span></span>
<span><span class="co">##  [81] blob_1.2.4                siggenes_1.80.0          </span></span>
<span><span class="co">##  [83] htmltools_0.5.8.1         scales_1.3.0             </span></span>
<span><span class="co">##  [85] png_0.1-8                 knitr_1.49               </span></span>
<span><span class="co">##  [87] tzdb_0.4.0                reshape2_1.4.4           </span></span>
<span><span class="co">##  [89] rjson_0.2.23              nlme_3.1-166             </span></span>
<span><span class="co">##  [91] curl_6.2.0                proxy_0.4-27             </span></span>
<span><span class="co">##  [93] cachem_1.1.0              rhdf5_2.50.2             </span></span>
<span><span class="co">##  [95] stringr_1.5.1             BiocVersion_3.20.0       </span></span>
<span><span class="co">##  [97] AnnotationDbi_1.68.0      restfulr_0.0.15          </span></span>
<span><span class="co">##  [99] GEOquery_2.74.0           pillar_1.10.1            </span></span>
<span><span class="co">## [101] grid_4.4.2                reshape_0.8.9            </span></span>
<span><span class="co">## [103] vctrs_0.6.5               xtable_1.8-4             </span></span>
<span><span class="co">## [105] evaluate_1.0.3            readr_2.1.5              </span></span>
<span><span class="co">## [107] GenomicFeatures_1.58.0    cli_3.6.3                </span></span>
<span><span class="co">## [109] compiler_4.4.2            Rsamtools_2.22.0         </span></span>
<span><span class="co">## [111] rlang_1.1.5               crayon_1.5.3             </span></span>
<span><span class="co">## [113] rngtools_1.5.2            labeling_0.4.3           </span></span>
<span><span class="co">## [115] nor1mix_1.3-3             mclust_6.1.1             </span></span>
<span><span class="co">## [117] plyr_1.8.9                stringi_1.8.4            </span></span>
<span><span class="co">## [119] BiocParallel_1.40.0       munsell_0.5.1            </span></span>
<span><span class="co">## [121] Matrix_1.7-1              hms_1.1.3                </span></span>
<span><span class="co">## [123] sparseMatrixStats_1.18.0  bit64_4.6.0-1            </span></span>
<span><span class="co">## [125] Rhdf5lib_1.28.0           KEGGREST_1.46.0          </span></span>
<span><span class="co">## [127] statmod_1.5.0             igraph_2.1.4             </span></span>
<span><span class="co">## [129] memoise_2.0.1             bslib_0.9.0              </span></span>
<span><span class="co">## [131] bit_4.5.0.1</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Victor Yuan, Wendy P. Robinson, Icíar Fernández-Boyano.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
